FSW/src/util/dataCollection.cpp

```diff
@@ -256,4 +262,61 @@ unsigned long calcTimestamp() {
     unsigned long timestamp = currentTimeRelative; 
 
     return timestamp;
+}
+
+/* - - - - - - downlink - - - - - - *
+ *  
+ * Usage:
+ *  downlinks all files on the flash module
+ * 
+ * Inputs:
+ *  none
+ *  
+ * Outputs:
+ *  none
+ */
+void downlink() {
+    
+    if (newDownlink) { // reset asynchronous variables if it is a new downlink
+        filename[FILE_IDX_OFFSET] = '0';
+        downlinkFileIdx = 0;
+        downlinkFileCount = 0;
+        newDownlink = false;
+    }
+
+    if (downlinkFileIdx < MAXFILES){ // check all science possible filenames
+ 
+        if (SerialFlash.exists(filename)) { // check if file exists
+            Serial.print("Downlinking ");
+            Serial.print(filename);
+            Serial.println(":");
+
+            char downlinkBuffer[FILESIZE];
+
+            // read data from file into downlink buffer
+            SerialFlashFile downlinkFile;
+            downlinkFile = SerialFlash.open(filename);
+            if (downlinkFile) {
+                downlinkFile.read(downlinkBuffer, FILESIZE);
+                Serial.println(downlinkBuffer);
```

Comment:
So right now the downlink just prints the data? That's fine, just wanted to make sure I understood this.

Reply:
`println` just adds a newline in at the end, I want these three prints all on the same line. Additionally, string concatenation can have issues on arduino so most examples I have seen use this syntax rather than concatenating the strings and using one `println` call.